cfpb/cfgov-refresh
# Add multiple location support to JobListingPages

This PR adds the ability to associate a job posting with multiple locations. Currently a JobListingPage can only be associated with either one office or one region; this change makes it so a job can be associated with either one or more offices or one or more regions.

This involved changes to a few different things:

1. Schema changes to the Django models used to manage location data (requiring 3 migrations to implement to make sure we have no data loss).
2. Modifying how the JobListingPage can be configured in the Wagtail admin and how it pulls that data back when the page gets rendered.
3. Support for proper rendering of all of the different combinations (one office, one office + remote, one region, multiple regions, etc) on both JobListingPages and associated StreamField blocks.

To assist with testing I also implemented a new TemplateDebugView that makes it easy to test our templates quickly with a set of simple test data.

Tagging backends for a review of the schema changes, mostly. Tagging designers for review of the styles in the screenshots section below. Also tagging @contolini because I can't request his review through GitHub.

## Testing

All tests assume use of a local server; this can also be tested internally on UCP.

Testing live pages:

1. Visit http://localhost:8000/about-us/careers/ to confirm that the job list there looks correct.
2. Visit http://localhost:8000/about-us/careers/current-openings/ to confirm that the job table there looks correct.
3. Visit a job opening, for example http://localhost:8000/about-us/careers/current-openings/consumer-response-specialist-investigations-221mpa/ and confirm that the job looks correct.

Testing the Wagtail admin:

1. Edit the sidebar of the Careers overview page at http://localhost:8000/admin/pages/3627/edit/#tab-sidebar and note the job list block.
2. Edit the Current Openings page at http://localhost:8000/admin/pages/4369/edit/ and note the job table block, which no longer has any options.
3. Edit a JobListingPage, for example http://localhost:8000/admin/pages/13832/edit/, and note the new interface.

If you experiment with changing the location on a job, you should be able to test various conditions that will fail page validation: no location data, specifying both offices and regions, checking the remote box for a region-based job.

You can explore the frontend variations at these template debug links:

- http://localhost:8000/admin/template_debug/jobmanager/job_listing_details/
- http://localhost:8000/admin/template_debug/jobmanager/job_listing_list/
- http://localhost:8000/admin/template_debug/jobmanager/job_listing_table/

## Screenshots

New location editing interface on JobListingPages:

![image](https://user-images.githubusercontent.com/654645/78912668-ddc97e00-7a55-11ea-891b-d834cceae5ff.png)

Frontend variations:

|Job listing details|Job listing list|Job listing table|
|-|-|-|
|![image](https://user-images.githubusercontent.com/654645/78912762-ff2a6a00-7a55-11ea-8b96-301df4fa7c1a.png)|![image](https://user-images.githubusercontent.com/654645/78912834-11a4a380-7a56-11ea-800b-7fa6fdfd424a.png)|![image](https://user-images.githubusercontent.com/654645/78912858-1bc6a200-7a56-11ea-8564-da470f6022c3.png)|

## Todos

- Per the CFPB style guide, months with short names like June should be displayed as "June", not abbreviated like "Jun.". This is a known issue and not something that will be addressed by this PR.
- The work on the TemplateDebugView is early; ideally this should be linked from the Wagtail sidebar somehow.

## Checklist
